server/filestore.go

```diff
@@ -453,31 +474,71 @@ func (fs *fileStore) genEncryptionKeys(context string) (aek cipher.AEAD, bek *ch
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	kek, err := chacha20poly1305.NewX(rb)
+
+	sc := fs.fcfg.Cipher
+
+	var kek cipher.AEAD
+	if sc == ChaCha {
```

Comment:
Looks like this block repeats 11 times in this PR, I think it justifies being made a function? Just pass the sc and rb.

Reply:
Will take a look.